### PR TITLE
gc_worker: limit the write flow of GC(#5263)

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

###  What have you changed?
cherry pick #5263

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -581,6 +592,7 @@ impl<E: Engine> TestStorageBuilder<E> {
         Storage::from_engine(
             self.engine,
             &self.config,
+            &GCConfig::default(),
```
Comment (MyonKeminta) [inline src/storage/mod.rs]: Maybe you need `self.gc_config` here?

Comment (youjiali1995) [reply src/storage/mod.rs]: yes

**Comment by youjiali1995** [comment]
@AndreMouche Please merge it manually.

**Pull request merged**